server/game/gameSystems/PlayCardSystem.ts

```diff
@@ -74,11 +103,12 @@ export class PlayCardSystem<TContext extends AbilityContext = AbilityContext> ex
     /**
      * Generate a play card ability for the specified card.
      */
-    private generatePlayCardAbility(card: Card, playType: PlayType) {
+    private generatePlayCardAbility(card: Card, playType: PlayType, costAdjusterForThisAction = null) {
+        const triggerHandlingMode = this.properties.nested ? TriggerHandlingMode.ResolvesTriggers : TriggerHandlingMode.PassesTriggersToParentWindow;
         switch (card.type) {
-            case CardType.BasicUnit: return new PlayUnitAction(card, playType, this.properties.entersReady);
-            case CardType.BasicUpgrade: return new PlayUpgradeAction(card, playType);
-            case CardType.Event: return new PlayEventAction(card, playType);
+            case CardType.BasicUnit: return new PlayUnitAction(card, playType, this.properties.entersReady, triggerHandlingMode, costAdjusterForThisAction);
```

Comment:
Just a heads-up this will have a merge conflict with the code Chebato just merged, he passes a property object to this method now so we can probably just append to that (if I'm remembering right)